### ISSUE TITLE
Run CI based on the check

### ISSUE THREAD

**Issue by Aarsh2001** [agilepathway/label-checker#297]
Can we run workflow files based on label checks?, if so, how?

**Comment by johnboyes** [comment]
Hi @Aarsh2001, thanks for raising this.  Can you elaborate a little please, and give me an example of what you would like to be able to do?  Thanks.

**Comment by Aarsh2001** [comment]
Sure, so i want my workflow files to run if there is a label present in the PR. So ideally i would like to first check if it is present, if it's true then run the jobs else skip.

Since, i was using `label:actions`, i found that it required for me to add a label for every commit in the PR. It triggered only once, when the label was added , it didn't  trigger for the next commit, unless it was added again.

In such a case, i wanted the `label checker` to check if the label is present and then run the jobs. Hope this clarifies?

**Comment by johnboyes** [comment]
That helps to clarify, thanks. A couple of initial questions/comments:

- > Since, i was using `label:actions`

   What's the `label:actions` that you are using, exactly?  I'm not familiar.

- My design philosophy for https://github.com/agilepathway/label-checker was to do one thing well:
   
   > check pull requests (PRs) for the presence or absence of specified labels
   
   By "check", I mean to prevent PRs from being merged unless the specified labels are absent/present.
   
   If I'm understanding your question right, you are asking more about _triggering_ workflows, right?  If that's the case, I wonder if https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#label is useful for you?
   
Feel free to come back if I've misunderstood!

**Comment by johnboyes** [comment]
Thinking about it more, I guess your use case is to have something very similar to this repo's `label-checker`, but which (instead of checking PRs) is a conditional that can be used inside a workflow to determine if the workflow should run.  That's an interesting idea if so, which raises the question of whether this `label-checker` could/should be extended to offer that.  My initial thoughts are that it would be best being a separate repo, as I'm a fan of the "do one thing and do it well" principle.  Thoughts?

**Comment by Aarsh2001** [comment]
Yes, you got it right. Also that we can create one workflow which checks for labels, and make other workflows depend on it, this worked in my case. I used the `label-checker.yml` and if it was a success other workflows ran. You can add this to the readme of the project maybe, as tricks and tips.😅

**Comment by johnboyes** [comment]
Nice!  Is your repo public?   Please share the link to the commit when you've implemented this if so, will be v useful to have a working example I can reference in the `README` 👍🏻.   If it's not a public repo then grateful if you can share a (sanitised) code snippet here, thanks 🙏🏻

**Comment by Aarsh2001** [comment]
Yes the repository is public, I'll share it after the implementation 🙂, also the code snippet!
